src/test/java/core/basesyntax/StoreStorageTest.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    @Test
+    public void storageTestBuyNothing() {
+        Storage storage = new Storage();
+        Operation buy = new Buy();
+        Operation supply = new Supply();
+        supply.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(SEVENTEENTH_OF_OCTOBER)), storage);
+        Assert.assertTrue(buy.updateStorage(new Product(BANANA, 0, LocalDate.parse(SEVENTEENTH_OF_OCTOBER)), storage));
+        List<String> expected = List.of(FIRST_OUTPUT_DATA
+                ,"Total amount:"
+                ,"banana,20");
+        List<String> actual = List.of(storage.toString().split("\n"));
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void storageTestBuyAllFruitsAreOverdue() {
+        Storage storage = new Storage();
+        Operation buy = new Buy();
+        Operation supply = new Supply();
+        supply.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(SEVENTEENTH_OF_OCTOBER)), storage);
+        Assert.assertFalse(buy.updateStorage(new Product(BANANA, 10, LocalDate.parse(EIGHTEENTH_OF_OCTOBER)), storage));
+        List<String> expected = List.of(FIRST_OUTPUT_DATA
+                ,"Total amount:"
+                ,"banana,20");
+        List<String> actual = List.of(storage.toString().split("\n"));
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void storageTestBuyFromOneBox() {
+        Storage storage = new Storage();
+        Operation buy = new Buy();
+        Operation supply = new Supply();
+        supply.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(SEVENTEENTH_OF_OCTOBER)), storage);
+        Assert.assertTrue(buy.updateStorage(new Product(BANANA, 10, LocalDate.parse(FIFTEENTH_OF_OCTOBER)), storage));
+        List<String> expected = List.of("banana,10,2020-10-17"
+                ,"Total amount:"
+                ,"banana,10");
+        List<String> actual = List.of(storage.toString().split("\n"));
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void storageTestBuyFullBox() {
+        Storage storage = new Storage();
+        Operation buy = new Buy();
+        Operation supply = new Supply();
+        supply.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(FIFTEENTH_OF_OCTOBER)), storage);
+        Assert.assertTrue(buy.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(TENTH_OF_OCTOBER)), storage));
+        Assert.assertEquals("", storage.toString());
+    }
+
+    @Test
+    public void storageTestBuyFromDifferentBox() {
+        Storage storage = new Storage();
+        Operation buy = new Buy();
+        Operation supply = new Supply();
+        supply.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(SEVENTEENTH_OF_OCTOBER)), storage);
+        supply.updateStorage(new Product(BANANA, TWENTY, LocalDate.parse(EIGHTEENTH_OF_OCTOBER)), storage);
+        Assert.assertTrue(buy.updateStorage(new Product(BANANA, 40, LocalDate.parse(FIFTEENTH_OF_OCTOBER)), storage));
```

Comment:
this line is longer than 100 chars fore sure, let's fix checkstyle in test files